server/sonar-server/src/main/java/org/sonar/server/plugins/ws/InstalledPluginsWsAction.java

```diff
@@ -78,21 +93,44 @@ public void handle(Request request, Response response) throws Exception {
       );
   }
 
-  private void writeMetadataList(Response response, Collection<PluginMetadata> pluginMetadatas) {
-    JsonWriter jsonWriter = response.newJsonWriter();
-    jsonWriter.beginArray();
-    for (PluginMetadata pluginMetadata : pluginMetadatas) {
-      writeMetadata(jsonWriter, pluginMetadata);
-    }
-    jsonWriter.endArray();
-    jsonWriter.close();
+  private void writeMetadataList(JsonWriter jsonWriter, Collection<PluginMetadata> pluginMetadatas) {
+      jsonWriter.name(OBJECT_PLUGINS);
+      jsonWriter.beginArray();
+      for (PluginMetadata pluginMetadata : pluginMetadatas) {
+        writeMetadata(jsonWriter, pluginMetadata);
+      }
+      jsonWriter.endArray();
   }
 
   private void writeMetadata(JsonWriter jsonWriter, PluginMetadata pluginMetadata) {
     jsonWriter.beginObject();
+
     jsonWriter.prop(PROPERTY_KEY, pluginMetadata.getKey());
     jsonWriter.prop(PROPERTY_NAME, pluginMetadata.getName());
+    jsonWriter.prop(PROPERTY_DESCRIPTION, pluginMetadata.getDescription());
     jsonWriter.prop(PROPERTY_VERSION, firstNonNull(pluginMetadata.getVersion(), DEFAULT_VERSION));
+    jsonWriter.prop(PROPERTY_LICENSE, pluginMetadata.getLicense());
+
+    writeOrganisation(jsonWriter, pluginMetadata);
+
+    writeUrls(jsonWriter, pluginMetadata);
+
+    jsonWriter.endObject();
+  }
+
+  private void writeOrganisation(JsonWriter jsonWriter, PluginMetadata pluginMetadata) {
+    jsonWriter.name(OBJECT_ORGANIZATION);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_NAME, pluginMetadata.getOrganization());
```

Comment:
The endless question about null versus empty strings. JsonWriter should still support the writing of empty strings, so I think about 2 options:
1. add a property JsonWriter#setIgnoreEmptyStrings(boolean). Default is false. The pb is that it will be applied to all field values.
2. convert empty to null in InstalledPluginsWsAction

I tend to prefer second option as long as there are no other use-cases.